src/software/embedded/robot_diagnostics_cli/embedded_data.py

```diff
@@ -16,43 +16,29 @@ class EmbeddedData:
 
     def __init__(self) -> None:
         # Initializes the redis cache connection
-        self.redis = redis.StrictRedis(
-            host=REDIS_DEFAULT_HOST,
-            port=REDIS_DEFAULT_PORT,
-            charset="utf-8",
-            decode_responses=True,
-        )
+        self.yaml_reader = yaml.safe_load(open(ROBOT_PATH_TO_YAML_CONFIG, "r"))
```

Comment:
can you edit this class to load a YAML file from the constructor? so that we can run the robot_diagnostics_cli on our computers with thunderloop without creating a `/home/robot/config.yaml` by specifying a config.yaml as a command line argument to `robot_diagnostics_cli`